pymc/printing.py

```diff
@@ -130,148 +93,176 @@ def str_for_model(model: Model, formatting: str = "plain", include_params: bool
+        args_str = f"f({args_str})"  # TODO do we still want to do this?
 
-def _str_for_input_var(var: Variable, formatting: str) -> str:
-    # Avoid circular import
-    from pymc.distributions.distribution import SymbolicRandomVariable
+    # These three strings are now formatted according to `formatting`. If latex, they
+    # just ultimately need to be wrapped in $.
+    return name, dist_name, args_str
 
-    def _is_potential_or_deterministic(var: Variable) -> bool:
-        try:
-            return var.str_repr.__func__.func is str_for_potential_or_deterministic
-        except AttributeError:
-            # in case other code overrides str_repr, fallback
-            return False
 
+def _str_for_input_var(var: Variable, formatting: str) -> str:
+    """Make a human-readable string representation for a variable that
+    serves as input to another variable."""
+    # Check for constants first, because they won't have var.owner
     if isinstance(var, (Constant, SharedVariable)):
-        return _str_for_constant(var, formatting)
-    elif isinstance(
-        var.owner.op, (RandomVariable, SymbolicRandomVariable)
-    ) or _is_potential_or_deterministic(var):
-        # show the names for RandomVariables, Deterministics, and Potentials, rather
-        # than the full expression
-        return _str_for_input_rv(var, formatting)
+        # Give the constant value if it's small enough, else basic type info.
+        # Previously _str_for_constant()
+        if isinstance(var, Constant):
+            var_data = var.data
+            var_type = "constant"
+        else:
+            var_data = var.get_value()
+            var_type = "shared"
+        if var_data.size == 1:
+            return f"{var_data.flatten()[0]:.3g}"
+        else:
+            return f"<{var_type} {var_data.shape}>"  # TODO shape info or nah?
     elif isinstance(var.owner.op, DimShuffle):
-        return _str_for_input_var(var.owner.inputs[0], formatting)
-    else:
-        return _str_for_expression(var, formatting)
-
-
-def _str_for_input_rv(var: Variable, formatting: str) -> str:
-    _str = (
-        var.name
-        if var.name is not None
-        else str_for_dist(var, formatting=formatting, include_params=True)
-    )
-    if "latex" in formatting:
-        return _latex_text_format(_latex_escape(_str.strip("$")))
-    else:
-        return _str
-
-
-def _str_for_constant(var: Union[Constant, SharedVariable], formatting: str) -> str:
-    if isinstance(var, Constant):
-        var_data = var.data
-        var_type = "constant"
-    else:
-        var_data = var.get_value()
-        var_type = "shared"
-
-    if len(var_data.shape) == 0:
-        return f"{var_data:.3g}"
-    elif len(var_data.shape) == 1 and var_data.shape[0] == 1:
-        return f"{var_data[0]:.3g}"
-    elif "latex" in formatting:
-        return rf"\text{{<{var_type}>}}"
+        # Recurse
+        return _str_for_input_var(var.owner.inputs[0], formatting=formatting)
```

Comment:
I carried this over from the previous version.
When would we receive a DimShuffle here, and can we be sure that only the first input is relevant?